modules/thermalmanager.c

```diff
@@ -52,545 +156,995 @@
+     *       be completely bogus when returning to more normal state
+     *       after having several sensors in elevated status.
+     *
+     *       Since consumers of this if should not care what is the
+     *       sensor that is reported, leaving this the way it has
+     *       always been...
+     */
 
-    dsme_log(LOG_DEBUG,
-             "thermalmanager: check thermal object '%s'",
-             thermal_object->conf->name);
-    thermal_object_polling_interval_expired(thermal_object);
-}
+    int         temperature = thermal_object_get_temperature(changed_object);
+    const char *sensor_name = thermal_object_get_name(changed_object);
 
+    thermal_manager_broadcast_status_dsme(status, temperature, sensor_name);
 
-DSME_HANDLER(DSM_MSGTYPE_DBUS_CONNECT, client, msg)
-{
-  dsme_log(LOG_DEBUG, "thermalmanager: DBUS_CONNECT");
-  dsme_dbus_bind_methods(&bound, methods, service, interface);
+EXIT:
+    return;
 }
 
-DSME_HANDLER(DSM_MSGTYPE_DBUS_DISCONNECT, client, msg)
+/** Broadcast overall device thermal status within dsme
+ *
+ * Map sensor thermal status to device status levels used by
+ * for example shutdown policy engine.
+ *
+ * Broadcast the changes via DSM_MSGTYPE_SET_THERMAL_STATUS event.
+ *
+ * @param status          device overall thermal status
+ * @param temperature     temperature to report
+ * @param sensor_name     sensor to report as cause of the change
+ */
+static void
+thermal_manager_broadcast_status_dsme(THERMAL_STATUS status,
+                                      int temperature,
+                                      const char *sensor_name)
 {
-  dsme_log(LOG_DEBUG, "thermalmanager: DBUS_DISCONNECT");
-  dsme_dbus_unbind_methods(&bound, methods, service, interface);
+    static dsme_thermal_status_t prev = DSM_THERMAL_STATUS_NORMAL;
+
+    /* Map sensor status to device status */
+    dsme_thermal_status_t curr = DSM_THERMAL_STATUS_NORMAL;
+
+    switch( status ) {
+    case THERMAL_STATUS_LOW:
+        curr = DSM_THERMAL_STATUS_LOWTEMP;
+        break;
+
+    default:
+    case THERMAL_STATUS_INVALID:
+    case THERMAL_STATUS_NORMAL:
+    case THERMAL_STATUS_WARNING:
+    case THERMAL_STATUS_ALERT:
+        break;
+
+    case THERMAL_STATUS_FATAL:
+        curr = DSM_THERMAL_STATUS_OVERHEATED;
+        break;
+    }
+
+    /* Skip broadcast if no change */
+    if( prev == curr )
+        goto EXIT;
+
+    prev = curr;
+
+    /* Log state change */
+    switch( curr ) {
+    case DSM_THERMAL_STATUS_LOWTEMP:
+        dsme_log(LOG_WARNING, PFIX"policy: low temperature (%s %dC)",
+                 sensor_name, temperature);
+        break;
+
+    case DSM_THERMAL_STATUS_NORMAL:
+        dsme_log(LOG_NOTICE, PFIX"policy: acceptable temperature (%s %dC)",
+                 sensor_name, temperature);
+        break;
+
+    case DSM_THERMAL_STATUS_OVERHEATED:
+        dsme_log(LOG_CRIT, PFIX"policy: overheated (%s %dC)", sensor_name,
+                 temperature);
+        break;
```

Comment:
`default: NOTREACHED;` (if we have such a macro or just warn unhandled/invalid state)